chia/consensus/block_body_validation.py

```diff
@@ -45,7 +46,7 @@ async def validate_block_body(
     height: uint32,
     npc_result: Optional[NPCResult],
     fork_point_with_peak: Optional[uint32],
-    get_block_generator: Callable,
+    get_block_generator: Callable[[BlockInfo], Awaitable[Optional[BlockGenerator]]],
```

Comment:
even though the function object we pass in here takes an extra (optional) parameter, we don't mention it in the type since we rely on it being optional within this function. Mypy seems fine converting the true function signature to this one

Reply:
If you are interested, you can hint argument names as well, though it is a bit more verbose.

```python
class SomeCallable(Protocol):
    def __call__(a_parameter: BlockInfo) -> Awaitable[Optional[BlockGenerator]]:
        pass
```

Reply:
I think you need `mypy --no-implicit-optional` to enforce that.